marimbabot_behavior/src/marimbabot_behavior/behavior_node.py

```diff
@@ -49,6 +49,19 @@ def __init__(self):
         while not self.lilypond_audio_client.wait_for_server(timeout=rospy.Duration(2)) and not rospy.is_shutdown():
             rospy.loginfo('Waiting for the audio_from_lilypond action server to come up')
 
+    # checks if the read note sequence includes a repeat symbol and updates the note sequence accordingly
+    def check_for_repeat(self):
+        if '\\repeat volta 2' in self.note_sequence:
+            # get the index of the first note
+            note_sequence_list = self.note_sequence.split(' ')
+            first_note_index = None
+            for i, x in enumerate(note_sequence_list):
+                if re.match(r'[a-g]\'*[0-9]+(\>)?(\.)?', x):
+                    first_note_index = i
+                    break
+            self.note_sequence = ' '.join(note_sequence_list[3:] + note_sequence_list[first_note_index:])
+            rospy.loginfo(f"updated notes: {self.note_sequence}")
```

Comment:
I think it would be better to use logdebug here?

Reply:
I use loginfo a lot in the behavior.. Whenever I modify the lilypond sequence, I like to immediately print the result. Should I replace it with logdebug wherever I print the (modified) note sequence (genuine question)?

Reply:
Normally, applications should be "as silent as possible" if everything is fine, as otherwise the terminal output grows in size very quickly for large applications. You can explicitly set certain nodes to a default loglevel or view their logs in rqt if needed.

Reply:
I see. I will change it then :+1: